docs/ff-concepts/adding-customization/common-examples.md

```diff
@@ -0,0 +1,284 @@
+
+```js
+Future addLocation(LatLng value) async {
+  // Add a new location to the LatLng list
+  FFAppState().addToLatLngList(value);
+}
+
+Future removeLocation(LatLng value) async {
+  // Remove a specific location from the LatLng list
+  FFAppState().removeFromLatLngList(value);
+}
+
+Future removeLocationAtIndex(int index) async {
+  // Remove a location at a specific index from the LatLng list
+  FFAppState().removeAtIndexFromLatLngList(index);
+}
+
+Future updateLocationAtIndex(int index, LatLng Function(LatLng) updateFn) async {
+  // Update a location at a specific index in the LatLng list
+  FFAppState().updateLatLngListAtIndex(index, updateFn);
+}
+
+Future insertLocationAtIndex(int index, LatLng value) async {
+  // Insert a new location at a specific index in the LatLng list
+  FFAppState().insertAtIndexInLatLngList(index, value);
+}
+```
+
+
+### Manipulate Custom Data Types
+When you create a custom data type in FlutterFlow, it **[generates a corresponding `<Name>Struct` class](../../generated-code/custom-data-type-gen.md)**. In FlutterFlow's custom code, you can create new instances of such data types, pass instances back into an action, or manipulate and retrieve information from existing objects. Here are some examples to help illustrate working with an example `ProductStruct` class.
+
+#### Example 1: Creating a new Instance of `ProductStruct`
+To create a new `ProductStruct` instance, initialize it with the required properties:
+
+```js
+// Create a new instance of ProductStruct
+final newProduct = ProductStruct(
+    productId: '123',
+    name: 'Example Product',
+    description: 'A sample product description.',
+    category: 'Electronics',
+    subCategory: 'Mobile Phones',
+    price: PriceStruct(amount: 299.99, currency: 'USD'),
+    sizes: ['Small', 'Medium', 'Large'],
+    colors: [ColorsStruct(colorName: 'Red', colorHex: '#FF0000')],
+    images: [ImagesStruct(thumbnail: 'https://example.com/image.jpg')],
+    stockStatus: StockStatusStruct(xs: 0, small: 2),
+    reviews: [ReviewsStruct(rating: 4, comment: 'Great product!')],
+);
+
+```
+
+#### Example 2: Accessing Properties of an Existing `ProductStruct` object
+
+If you have an existing `ProductStruct` object (e.g., retrieved from a list of products), you can access its properties or return specific values back to the calling Action. Let's assume you have an Action that calls a Custom Action to retrieve a field value from the provided `ProductStruct` object.
+
+- **Returning a Single Field from ProductStruct**
+
+This function retrieves and returns the product's name. The return type is `String?` to account for the possibility of a null value.
+```js
```

Comment:
Also call out that all these code samples seem to be in js but I think dart is supported?

Reply:
Dart is not supported in Docusaurus's code highlighters :( 
So if I do "```dart" it wont highlight the code at all. Also in the rendered docs, they wont be able to see the language of the code, just a copy code button is present. So we won't need to callout necessarily.